scripts/auto-rebase/rebase.py

```diff
@@ -0,0 +1,317 @@
+            sys.exit(f"Could not get environment variable '{var_name}'")
+        else:
+            logging.info(f"Could not get environment variable '{var_name}' - ignoring")
+            return ""
+    return val
+
+
+def run_rebase_sh(release_amd64, release_arm64):
+    script_dir = os.path.abspath(os.path.dirname(__file__))
+    args = [f"{script_dir}/rebase.sh", "to", release_amd64, release_arm64]
+    logging.info(f"Running: '{' '.join(args)}'")
+    result = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, universal_newlines=True)
+    logging.info(f"Output:\n\n{result.stdout}\n------------------------------\n")
+    logging.info(f"Script returned code: {result.returncode}")
+    rr = namedtuple("RebaseResult", ["success", "output"])
+    return rr(success=result.returncode == 0, output=result.stdout)
+
+
+def commit_str(commit):
+    return f"{commit.hexsha[:8]} - {commit.summary}"
+
+
+def get_installation_access_token(app_id, key_path, org, repo):
+    integration = GithubIntegration(app_id, Path(key_path).read_text())
+    app_installation = integration.get_installation(org, repo)
+    if app_installation == None:
+        sys.exit(f"Failed to get app_installation for {org}/{repo}. Response: {app_installation.raw_data}")
+    return integration.get_access_token(app_installation.id).token
+
+
+def make_sure_rebase_script_created_new_commits_or_exit(git_repo, base_branch):
+    if git_repo.active_branch.commit == git_repo.branches[base_branch].commit:
+        logging.info(f"There's no new commit on branch {git_repo.active_branch} compared to '{base_branch}' "
+                     "meaning that the rebase.sh script didn't create any commits and "
+                     "MicroShift is already rebased on top of given release.\n"
+                     f"Last commit: {commit_str(git_repo.active_branch.commit)}")
+        sys.exit(0)
+
+
+def get_remote_with_token(git_repo, token, org, repo):
+    remote_url = f"https://x-access-token:{token}@github.com/{org}/{repo}"
+    try:
+        remote = git_repo.remote(BOT_REMOTE_NAME)
+        remote.set_url(remote_url)
+    except ValueError:
+        git_repo.create_remote(BOT_REMOTE_NAME, remote_url)
+
+    return git_repo.remote(BOT_REMOTE_NAME)
+
+
+def try_get_rebase_branch_ref_from_remote(remote, branch_name):
+    remote.fetch()
+    matching_remote_refs = [ref for ref in remote.refs if BOT_REMOTE_NAME + "/" + branch_name == ref.name]
+
+    if len(matching_remote_refs) == 0:
+        logging.info(f"Branch '{branch_name}' does not exist on remote")
+        return None
+
+    if len(matching_remote_refs) > 1:
+        matching_branches = ", ".join([r.name for r in matching_remote_refs])
+        logging.warning(f"Found more than one branch matching '{branch_name}' on remote: {matching_branches}. Taking first one")
+        _extra_msgs.append(f"Found more than one branch matching '{branch_name}' on remote: {matching_branches}.")
+        return matching_remote_refs[0]
```

Comment:
Are the duplicate branches returned in a deterministic order?

Reply:
Refs must match exactly `remote/branch-name` and you can't have two branches named the same so I don't think this would ever happen but wanted to check all possibilities just in case.